## Pandas With Alishba


-----

> for life (I'm In............)

`Sok lewanai shi os sok lewantoob shi`\
`sok shi pa khob oda sok rangeen khob she`

`Zama Zor zama Zwani sta pa Husan Qurbani`

` A^2 `

------------------

**Importing the **libraries****

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

import dataset of Gender based Employment in Pakistan

In [6]:
df = pd.read_csv("D:/coding files/AK CODE/practice_sessions/EDA/Gender based employment in pakistan/GenderBasedEmploymentInPakistan2023.csv")

In [7]:
df

Province            Division District               Indicator  \
0     Balochistan      Kalat Division    Kalat  Working Age Population   
1     Balochistan      Kalat Division    Kalat  Working Age Population   
2     Balochistan      Kalat Division    Kalat            Labour Force   
3     Balochistan      Kalat Division    Kalat            Labour Force   
4     Balochistan      Kalat Division    Kalat                Employed   
...           ...                 ...      ...                     ...   
4725        Sindh  Hyderabad Division   Matiri        Total Population   
4726        Sindh   Banbhore Division  Thattah        Total Population   
4727        Sindh   Banbhore Division  Thattah        Total Population   
4728  Balochistan     Quetta Division   Lorali        Total Population   
4729  Balochistan     Quetta Division   Lorali        Total Population   

     Area Type   Total    Male  Female  
0        Rural  130000   70000   59000  
1        Urban   28000   17000   12000  
2        Rural   52600   49600    3100  
3        Urban   10800   10600     300  
4        Rural   49700   47400    2300  
...        ...     ...     ...     ...  
4725     Urban  182669   94445   88207  
4726     Rural  654818  340701  314105  
4727     Urban  176476   90900   85556  
4728     Rural  332532  176186  156332  
4729     Urban   64891   36316   28573  

[4730 rows x 8 columns]

In [8]:
df.head()

Province        Division District               Indicator Area Type  \
0  Balochistan  Kalat Division    Kalat  Working Age Population     Rural   
1  Balochistan  Kalat Division    Kalat  Working Age Population     Urban   
2  Balochistan  Kalat Division    Kalat            Labour Force     Rural   
3  Balochistan  Kalat Division    Kalat            Labour Force     Urban   
4  Balochistan  Kalat Division    Kalat                Employed     Rural   

    Total   Male Female  
0  130000  70000  59000  
1   28000  17000  12000  
2   52600  49600   3100  
3   10800  10600    300  
4   49700  47400   2300

In [9]:
df.tail(6)

Province            Division District         Indicator Area Type  \
4724        Sindh  Hyderabad Division   Matiri  Total Population     Rural   
4725        Sindh  Hyderabad Division   Matiri  Total Population     Urban   
4726        Sindh   Banbhore Division  Thattah  Total Population     Rural   
4727        Sindh   Banbhore Division  Thattah  Total Population     Urban   
4728  Balochistan     Quetta Division   Lorali  Total Population     Rural   
4729  Balochistan     Quetta Division   Lorali  Total Population     Urban   

       Total    Male  Female  
4724  587371  302477  284834  
4725  182669   94445   88207  
4726  654818  340701  314105  
4727  176476   90900   85556  
4728  332532  176186  156332  
4729   64891   36316   28573

In [10]:
df.shape

(4730, 8)

It has just told us that theres is 4730 rows and 8 columns

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4730 entries, 0 to 4729
Data columns (total 8 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Province   4730 non-null   object
 1   Division   4730 non-null   object
 2   District   4730 non-null   object
 3   Indicator  4730 non-null   object
 4   Area Type  4730 non-null   object
 5   Total      4730 non-null   object
 6   Male       4730 non-null   object
 7   Female     4730 non-null   object
dtypes: object(8)
memory usage: 295.8+ KB


we can see that all the columns have datatype of object which should not be .\
we will work on that 

In [13]:
df.describe()

Province         Division        District         Indicator Area Type  \
count      4730             4730            4730              4730      4730   
unique        4               30             131                20         3   
top      Punjab  Quetta Division  Qilla Abdullah  Total Population     Rural   
freq       1334              400              38               276      2103   

       Total  Male Female  
count   4730  4730   4730  
unique  1838  1848   1626  
top        -     -      -  
freq     322   343    742

study the above

-----

we will be changing the datatypes of "total", "Male","Female"

changing the datatypes

In [18]:
df.Male.unique()

array(['70000', '17000', '49600', ..., '90900', '176186', '36316'],
      dtype=object)

In [27]:
rm1 = df[(df["Male"]=="-")]

In [31]:
df = df.drop(df[(df["Male"]=="-")])

KeyError: "['Province', 'Division', 'District', 'Indicator', 'Area Type', 'Total', 'Male', 'Female'] not found in axis"